crates/compiler/gen_llvm/src/llvm/lowlevel.rs

```diff
@@ -2059,6 +2059,54 @@ fn dec_unary_op<'ctx>(
     }
 }
 
+fn dec_binary_op<'ctx>(
+    env: &Env<'_, 'ctx, '_>,
+    fn_name: &str,
+    dec1: BasicValueEnum<'ctx>,
+    dec2: BasicValueEnum<'ctx>,
+) -> BasicValueEnum<'ctx> {
+    use roc_target::Architecture::*;
+    use roc_target::OperatingSystem::*;
+
+    let dec1 = dec1.into_int_value();
+    let dec2 = dec2.into_int_value();
+
+    match env.target_info {
+        TargetInfo {
+            architecture: X86_64 | X86_32,
+            operating_system: Unix,
+        } => {
+            let (low1, high1) = dec_split_into_words(env, dec1);
+            let (low2, high2) = dec_split_into_words(env, dec2);
+            let lowr_highr = call_bitcode_fn(
+                env,
+                &[low1.into(), high1.into(), low2.into(), high2.into()],
```

Comment:
Just curious, with the llvm fixes, will this go away? I think so right? Just a single i128?

Reply:
I think we need the next version of zig to actually pass `i128` instead of `{ i64, i64 }` for this to really go away